vyper/compiler/settings.py

```diff
@@ -12,3 +14,31 @@
     VYPER_TRACEBACK_LIMIT = int(_tb_limit_str)
 else:
     VYPER_TRACEBACK_LIMIT = None
+
+
+class OptimizationLevel(Enum):
+    NONE = 1
+    GAS = 2
+    CODESIZE = 3
+
+    @classmethod
+    def from_string(cls, val):
+        match val:
+            case "none":
+                return cls.NONE
+            case "gas":
+                return cls.GAS
+            case "codesize":
+                return cls.CODESIZE
+        raise ValueError(f"unrecognized optimization level: {val}")
+
+    @classmethod
+    def default(cls):
+        return cls.GAS
+
+
+@dataclass
+class Settings:
+    compiler_version: Optional[str] = None
+    optimize: Optional[OptimizationLevel] = None
```

Comment:
```suggestion
    optimize: OptimizationLevel = OptimizationLevel.default
```

Reply:
discussed offline - need the None sentinel to indicate whether the option was provided or not